frontend/src/metabase/admin/datamodel/components/filters/pickers/DatePicker/DatePickerFooter.tsx

```diff
@@ -62,7 +62,7 @@ const DatePickerFooter: React.FC<React.PropsWithChildren<Props>> = ({
         operator,
         field,
         start,
-        operator === "between" && !end ? start : end,
+        ...(operator === "between" ? [end ?? start] : []),
```

Comment:
This is raw MBQL. Yes, dashboard components use raw MBQL for state and then convert it to string values; it's not used for MBQL queries in dashboards. The unit test verifies that we get correct raw MBQL now.